src/rpc/rawtransaction.cpp

```diff
@@ -47,6 +48,32 @@
  */
 static const CFeeRate DEFAULT_MAX_RAW_TX_FEE_RATE{COIN / 10};
 
+bool checkIfStakeUsedInWithdrawalTransaction(const CStakesDbEntry& stake)
+{
+    assert(stake.isValid());
+    return !::ChainstateActive().CoinsTip().HaveCoin(COutPoint{stake.getKey(), stake.getNumOutput()});
+}
+
+std::string getTransactionType(const CTransaction& tx)
+{
+    uint256 hash{tx.GetHash()};
+    const CStakesDbEntry& stake = ::ChainstateActive().GetStakesDB().getStakeDbEntry(hash);
+    if(stake.isValid() && checkIfStakeUsedInWithdrawalTransaction(stake)) {
+        return "STAKING_WITHDRAWAL";
+    }
```

Comment:
`Staking withdrawal` transaction is a transaction in which one or more stakes are spent. To determine a transaction like this, one should iterate over all transaction inputs and check if any of them was a stake.